FunctionalArray.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  FunctionalArray.swift
+//  SwiftTrainingExercise
+//
+//  Created by Ahmed Ibrahim on 20/06/2023.
+//
+
+import Foundation
+
+class FunctionalArray<Element> where Element: Equatable & Comparable & Hashable {
```

Comment:
the "where" should be used with extensions
as we use to enable special functions when the cases is matched, in the way above the class won't work normally